### ISSUE TITLE
Color data browser cells by data type

### ISSUE THREAD

**Issue by gitblit** [sqlitebrowser/sqlitebrowser#2318]
Describe the new feature
--------------------------

HeidiSQL has a very nice feature that colors data browser cell foreground by the data type.  DB4S has something similar AFAICT for null and binary, but perhaps it could be extended to support other types?

- text: green
- integer: blue
- temporal: dark red/maroon
- others, no doubt

This makes it very easy to identify data types, helps mark your place when scrolling tables, and facilitates discovery when presented with a new schema/database.  After using this for many years, it's hard to go back to plain data grids.

Additional info
---------------
Please answer these questions before submitting your feature request.

#### Is your feature request related to an issue? Please include the issue number.

I didn't notice one.

#### Does this feature exist in another product or project? Please provide a link.

Yes.  https://www.heidisql.com/screenshots.php?which=data

#### Do you have a screenshot? Please add screenshots to help explain your idea.

![image](https://user-images.githubusercontent.com/868217/86264500-99e7a000-bb90-11ea-8ba3-8163e2c25fb3.png)

**Comment by chrisjlocke** [comment]
As a horrible unworkable and totally unfeasible hack, it is my unwritten duty to point out that you could probably do this using conditional formats (right-click a cell and select 'conditional formats')

![image](https://user-images.githubusercontent.com/7872218/86270455-29a64200-bbc3-11ea-8e44-ef5661f2d075.png)

You'd have to do this for each column in each table though, so would be unworkable.  
This is useful though for columns (for example, your usertype above) to identify columns that have a zero value or a value over 50.

**Comment by gitblit** [comment]
That is a very cool feature!  Thanks for bringing it to my attention.  As managed currently, I agree with you - unworkable.

> the only thing it gives is a bit of colour to an otherwise monochrome view of the data

That would be the main point.  The conditional formatting is way more powerful but not scalable.

> Its also slightly false where 'decimalFormat' is green, suggesting text when it's numerical

Hah!  I left the defaults.  :)  The variation/contrast is what I find appealing.  Then again I also like the RainbowCSV plugin for VS Code so I might be in the minority of thinking this is a grand idea.  :P

**Comment by chrisjlocke** [comment]
As another 'fly in the ointment', I know with some of the projects I've written, adding colour can slow down the display of the data.  Instead of just slapping text in a cell, it then has to go, "aah, lets now make that red" and "lets make that green".  If you've 3000 rows, it can add valuable seconds to the display.  Having said that, that's using the default controls in Visual Studio, rather than third party controls, which can be a tad superior.  So this post is just waffle.  🧇  (apparently, that's a waffle emoji!)

**Comment by justinclift** [comment]
Nah, as long as the code to set the colours is reasonable there shouldn't be any perceptible speed difference.  This is C++ and Qt, not MS Visual Basic. :stuck_out_tongue:
